packages/types/src/interfaces/contracts/definitions.ts

```diff
@@ -45,7 +45,7 @@ export default {
       origin: 'AccountId',
       dest: 'AccountId',
       value: 'Balance',
-      gasLimit: 'u64',
+      gasLimit: 'Option<Weight>',
```

Comment:
This is going to be problematic - there is now old and new, we need to cater for both.

Reply:
does `u64 | Option<Weight>` work?

Reply:
For his one we need a runtime result for V1 as well - that keeps as it. For v2, we have a new version that gets returned. So basically rename the existing to `ContractCallRequestV1` (to be returned for v1 runtime) and the change the above version for `Option<Weight>`.

Basically (until we finally have all this info on the metadata), we need to have old and new types available. Yes, it is a bitch.

Reply:
I already created `ContractCallRequestV1`, on line 44